glotaran/analysis/problem.py

```diff
@@ -209,7 +209,7 @@ def full_penalty(self) -> np.ndarray:
 
     @property
     def cost(self) -> float:
-        return np.sum(self._full_penalty)
+        return 0.5 * np.dot(self.full_penalty, self.full_penalty)
```

Comment:
See [scipy](https://github.com/scipy/scipy/blob/5b3a3b440525aa5038e2cead1f9b4f23a610cd9f/scipy/optimize/_lsq/least_squares.py#L79) also self.full_penalty is needed, self._full_penalty can be None (race condition)

Reply:
fine with me